internal/operator/operator.go

```diff
@@ -119,13 +125,20 @@ func (o Operator) Start(cfg config.Config) error {
 		MapperProvider: func(c *rest.Config, httpClient *http.Client) (meta.RESTMapper, error) {
 			return apiutil.NewDiscoveryRESTMapper(c, httpClient)
 		},
-		Cache:                  cache.Options{Namespaces: cfg.ScanNamespaces},
-		MetricsBindAddress:     metricsAddr,
+		Metrics:                metricsOpts,
 		HealthProbeBindAddress: probeAddr,
 		LeaderElection:         enableLeaderElection,
 		LeaderElectionID:       "398aa7bc.statnett.no",
 	}
 
+	if len(cfg.ScanNamespaces) > 0 {
+		watched := make(map[string]cache.Config)
+		for _, n := range cfg.ScanNamespaces {
+			watched[n] = cache.Config{}
+		}
+		options.Cache.DefaultNamespaces = watched
+	}
```

Comment:
Tempting to do

```go
		Cache: cache.Options{
			DefaultNamespaces: make(map[string]cache.Config),
		},
```

and do the loop regardless:

```go
	for _, n := range cfg.ScanNamespaces {
		options.Cache.DefaultNamespaces[n] = cache.Config{}
	}
```

---

Alternatively, to not get an empty map if list is empty

```go
	for _, n := range cfg.ScanNamespaces {
		if options.Cache.DefaultNamespaces == nil {
			 options.Cache.DefaultNamespaces = make(map[string]cache.Config)
		}
		options.Cache.DefaultNamespaces[n] = cache.Config{}
	}
```

Reply:
@tenstad Updated! PTAL!